hstream/src/HStream/Server/Core/Stream.hs

```diff
@@ -42,13 +49,17 @@ import           HStream.Utils
 -- but return failure response
 createStream :: HasCallStack => ServerContext -> API.Stream -> IO (ServerResponse 'Normal API.Stream)
 createStream ServerContext{..} stream@API.Stream{
-  streamBacklogDuration = backlogSec, ..} = do
+  streamBacklogDuration = backlogSec, streamShardCount = shardCount, ..} = do
   when (streamReplicationFactor == 0) $ throwIO (InvalidArgument "Stream replicationFactor cannot be zero")
   let streamId = transToStreamName streamStreamName
       attrs = S.def{ S.logReplicationFactor = S.defAttr1 $ fromIntegral streamReplicationFactor
                    , S.logBacklogDuration   = S.defAttr1 $
                       if backlogSec > 0 then Just $ fromIntegral backlogSec else Nothing}
   catch (S.createStream scLDClient streamId attrs) (\(_ :: S.EXISTS) -> throwIO StreamExists)
+  let shards = if shardCount <= 0 then 1 else shardCount
+  let partions :: [Word32] = [0..shards - 1]
+  forM_ partions $ \idx -> do
+    S.createStreamPartition scLDClient streamId (Just . getShardName $ fromIntegral idx)
```

Comment:
NOTE: This should be considered an inefficient action.

Reply:
Yep, But if we delay creating the shard until we need to write data, we may encounter the "newly created log cannot be written immediately" error.